setup.py

```diff
@@ -13,6 +13,7 @@
 
 REQUIREMENTS_TEST = open(os.path.join(HERE, "requirements_test.txt")).readlines()
 REQUIREMENTS = [
+    "smart_open",
```

Comment:
Could you change this to `smart_open[http]`? We already depend on requests, but I think it's "more correct" to explicitly specify the dependency. On that note, could you also add the other extras to the `EXTRAS` lists below?

Reply:
would you like to introduce s3 in EXTRAS? my minimally invasive suggestion would be to add s3,http extras here, and add gcs&azure extras to the EXTRAS

Reply:
Yep, that sounds good 👍

Reply:
done in #304 (and only http extra in this PR to strictly separate the features)